lib/downloader.rb

```diff
@@ -0,0 +1,35 @@
+# Defines how chromebrew downloads files.
+require_relative 'color'
+
+def notimplemented(package)
+  puts "#{package} has not been implemented as a downloader for Chromebrew yet. Defaulting to curl.".orange
+  @_crew_downloader = "curl -#LC -"
+  def download_file(url, filename)
+    system "#{@_crew_downloader} \'#{url}\' --output #{filename}"
+  end
+end
+
+case ENV["CREW_DOWNLOADER"]
+when 'aria2', 'aria2c'
+  def download_file(url, filename)
+    system "aria2c -c \'#{url}\' --out #{filename}"
+  end
+when 'curl', '', nil
+  def download_file(url, filename)
+    system "curl -#LC - \'#{url}\' --output #{filename}"
+  end
+when 'wget'
+  def download_file(url, filename)
+    system "wget -c \'#{url}\' --output-document #{filename}"
+  end
+when 'system curl', 'system'
+  def download_file(url, filename)
+    system "/usr/bin/curl -#LC - \'#{url}\' --output #{filename}"
+  end
+when 'fetch', 'bsdfetch', 'libfetch'
+  notimplemented 'BSD Fetch'
+when 'httpie'
+  notimplemented 'HTTPie'
+else
+  abort "CREW_DOWNLOADER is set to an unknown value. Exiting.".lightred
+end
```

Comment:
Maybe add a rescue block that will retry if the download command above failed:
```ruby
@retry_times = 0
unless $?.exitstatus == 0
  if @retry_times == 3
    abort 'Download failed :/'.lightred
  end
  puts "Downloader exited with status #{$?.exitstatus}, retrying…".yellow
  @retry_times += 1
  download_file(url, filename)
end
```

Reply:
If I recall correctly, this is built into every downloader we're using except fetch.

Reply:
You’re right but the `—retry` option in `curl` will not handle some server errors.